test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Utility/FileWatchers/SolutionConfigFileWatcherTests.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Threading;
+using FluentAssertions;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio.Utility.FileWatchers;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test.Utility.FileWatchers
+{
+    public class SolutionConfigFileWatcherTests
+    {
+        [Fact]
+        public void FileChanged_WhenFileCreated_NotificationReceived()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            using SolutionConfigFileWatcher target = new(testDirectory.Path);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            target.FileChanged += (s, e) => mre.Set();
+            string configPath = Path.Combine(testDirectory.Path, Settings.DefaultSettingsFileName);
+
+            // Act
+            File.WriteAllText(configPath, string.Empty);
+            // filesystem watchers are not real time, so we need to give a litle time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Fact]
+        public void FileChanged_WhenFileModified_NotificationReceivedAsync()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string configPath = Path.Combine(testDirectory.Path, Settings.DefaultSettingsFileName);
+            File.WriteAllText(configPath, string.Empty);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            using SolutionConfigFileWatcher target = new(testDirectory.Path);
+            target.FileChanged += (s, e) => mre.Set();
+
+            // Act
+            mre.Wait(0).Should().BeFalse();
```

Comment:
nit: Seems like an Arrange to me. (same applies to multiple copies of this line)

Reply:
Could it be 1 instead of 0 to avoid random runtime flakiness just in case?